### PR TITLE
Feature contact form validation more feedback

### PR DESCRIPTION
**Pull request by tnathu-ai**
contact form with instant feedback, 
contact purpose is not a required field

### CODE REVIEW

**Review by rider3458 — APPROVED** [review]

File: public/js/contact/contact.js [hunk 1]
```diff
@@ -76,26 +87,23 @@ FORM.addEventListener("submit", function () {
...
+    console.log(errNum);
 
     if (errNum>0) {
         // If errNum is greater than 0 than alert error and return "false"
         alert(err);
         return false;
     } else {
         // If errNum is less than 0 or 0 than alert "done" and return "true"
-        alert("Sent!");
+        alert("Your contact from successfully submitted!");
```
Comment (rider3458) [inline public/js/contact/contact.js:98]: contact form

**Review by rider3458 — APPROVED** [review]

File: public/mall/contact/index.html [hunk 2]
```diff
@@ -13,230 +13,255 @@
...
 
-          <div class="name">
-            <label for="lname" class="title required">Last Name</label>
-            <input id="lname" type="text" name="lname" class="name-field" placeholder="This field is required" required>
+        <div class="name">
+          <label for="lname" class="title required">Last Name</label>
+          <input id="lname" type="text" name="lname" class="name-field" placeholder="This field is required" required>
+          <div class="error-message" id="message-lname">
+            <h6>Last name must contain the following:</h6>
+            <p class="name-greater-3 invalid" id="lname-greater-3">First name contains <b> at least 3 </b> letters</p>
```
Comment (doanyennhi) [inline public/mall/contact/index.html:112]: Should be "Last name contains"

File: public/mall/contact/index.html [hunk 3]
```diff
@@ -13,230 +13,255 @@
...
           </div>
         </div>
+      </div>
+
+      <label for="email_add" class="title required">Email</label>
+        <span class="requirement-text">Ex: 123@1.me or 1.2.3@domain.name or this.is.my.email@sub.sub.123.main</span>
+      <input id="email_add" type="email" name="email_add" class="field" placeholder="This field is required" required>
+      <div class="error-message" id="message-email">
+        <h6>Last name must contain the following:</h6>
```
Comment (doanyennhi) [inline public/mall/contact/index.html:121]: Should be "Email must contain"

File: public/js/contact/contact.js [hunk 4]
```diff
@@ -76,26 +87,23 @@ FORM.addEventListener("submit", function () {
...
+    console.log(errNum);
 
     if (errNum>0) {
         // If errNum is greater than 0 than alert error and return "false"
         alert(err);
         return false;
     } else {
         // If errNum is less than 0 or 0 than alert "done" and return "true"
-        alert("Sent!");
+        alert("Your contact form successfully submitted!");
```
Comment (doanyennhi) [inline public/js/contact/contact.js:98]: Should be "Your contact form is successfully submitted!"

**Review by doanyennhi — CHANGES_REQUESTED** [review]
For name validation: when entering characters that are not letters, it still passes the validation check

For email validation:
- The requirement for length of [name] is at least 3. However, when the length of [name] is 1 or 2, it still passes the validation check
- The part of [domain] after the last dot has max length of 5, but when the length >5,  it still passes the validation check

For phone validation:
- Valid phone number has 9-11 digits. However, when I try number with 9 digits, it doesnt pass the validation check. 
- When I try number with 12 digits, it still passes validation check

**Review by miketvo — APPROVED** [review]
Well done!

**Pull request merged**